doc/oper.xml

```diff
@@ -1964,3 +1964,157 @@ gap> ModularProduct(NullDigraph(0), CompleteDigraph(10));
+  AncestorFunc and/or CrossFunc of the ExecuteDFS function.
+  See <Ref Func="ExecuteDFS" Label="for a record, object, integer and 4 functions"/>.
+  <Example><![CDATA[
+gap> PreOrderFunc := function(record, data)
+>       data.num_vertices := data.num_vertices + 1;
+>    end;;
+gap> record := NewDFSRecord(CompleteDigraph(2));;
+gap> data := rec(num_vertices := 0);;
+gap> ExecuteDFS(record, data, 1, PreOrderFunc,
+>               DFSDefault, DFSDefault, DFSDefault);
+gap> data;
+rec( num_vertices := 2 )
+gap> record := NewDFSRecord(CompleteDigraph(2));;
+gap> ExecuteDFS(record, [], 1, DFSDefault,
+>               DFSDefault, DFSDefault, DFSDefault);
+gap> record;
+rec( child := 1, current := 1, 
+  graph := <immutable complete digraph with 2 vertices>, 
+  parent := [ 1, 1 ], postorder := [ 2, 1 ], preorder := [ 1, 2 ], 
+  stop := false )
+]]></Example>
+</Description>
+</ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="ExecuteDFS">
+<ManSection>
+  <Oper Name="ExecuteDFS" Arg="record, data, start, PreOrderFunc, PostOrderFunc, AncestorFunc,
+                               CrossFunc"/>
+  <Description>
+  This performs a full depth first search from the <A>start</A> vertex (where <A>start</A> is within the graph).
+  The depth first search can be terminated by changing the <A>record</A>.stop attribute to true in the
+  <A>PreOrderFunc</A>, <A>PostOrderFunc</A>, <A>AncestorFunc</A> or <A>CrossFunc</A> functions.
+  ExecuteDFS takes 7 arguments:
+  <List>
+    <Item><A>record</A>: the depth first search record (created using NewDFSRecord)</Item>
+    <Item><A>data</A>: an object that you want manipulate in the functions passed.</Item>
+    <Item><A>start</A>: the vertex where we begin the depth first search.</Item>
+    <Item><A>PreOrderFunc</A>: this function is called when a vertex is first visited. This vertex
+    is stored in <A>record</A>.current</Item>
+    <Item><A>PostOrderFunc</A>: this function is called when a vertex has no more unvisited children
+    causing us to backtrack. This vertex is stored in <A>record</A>.child and its parent is stored 
+    in <A>record</A>.current</Item>
+    <Item><A>AncestorFunc</A>: this function is called when (<A>record</A>.current,
+    <A>record</A>.child) is an edge and <A>record</A>.child is an ancestor of <A>record</A>.current. An ancestor here means that
+    <A>record</A>.child is on the same branch as <A>record</A>.current but was visited prior to <A>record</A>.current</Item>
+    <Item><A>CrossFunc</A>: this function is called when (<A>record</A>.current,
+    <A>record</A>.child) is an edge and <A>record</A>.child has been visited before <A>record</A>.current
+      and it is not an ancestor of <A>record</A>.current</Item>
+  </List>
+  Note that this function only performs a depth first search on the connected graph that has <A>start</A>.
+  To perform a depth first search on unconnected components, you need to call this function with a <A>start</A> that is
```

Comment:
```suggestion
  To perform a depth first search on all connected components of the digraph, you need to call this function with a <A>start</A> that is
```
I feel that the term "unconnected component" is not used in graph theory. We mainly refer to the components as connected components (despite that here you mean components that contain verts that are not connected to the start vertex). Again if DFS is executed just as in dominators, I feel that we shouldn't talk about connected components but about reachability of vertices from the start vertex and something equivalent to connected components for this. @james-d-mitchell any views on this? :)

Reply:
This is certainly sensible. Making the change now.

Reply:
@james-d-mitchell